File: app/src/main/java/co/tpcreative/suppersafe/ui/lockscreen/CustomizationOptionsBundle.java
package co.tpcreative.suppersafe.ui.lockscreen;

import android.graphics.drawable.Drawable;

/**
 * The customization options for the buttons in {@link PinLockView}
 * passed to the {@link PinLockAdapter} to decorate the individual views
 *
 * Created by aritraroy on 01/06/16.
 */
public class CustomizationOptionsBundle {

    private int textColor;
    private int textColorVerify;
    private int textSize;
    private int buttonSize;
    private Drawable buttonBackgroundDrawable;
    private Drawable verifyButtonDrawable;
    private int verifyButtonWidthSize, verifyButtonHeightSize;
    private boolean showVerifyButton;
    private int verifyButtonPressesColor;

    public CustomizationOptionsBundle() {
    }

    public int getTextColorVerify() {
        return textColorVerify;
    }

    public void setTextColorVerify(int textColorVerify) {
        this.textColorVerify = textColorVerify;
    }

    public int getTextColor() {
        return textColor;
    }

    public void setTextColor(int textColor) {
        this.textColor = textColor;
    }

    public int getTextSize() {
        return textSize;
    }

    public void setTextSize(int textSize) {
        this.textSize = textSize;
    }

    public int getButtonSize() {
        return buttonSize;
    }

    public void setButtonSize(int buttonSize) {
        this.buttonSize = buttonSize;
    }

    public Drawable getButtonBackgroundDrawable() {
        return buttonBackgroundDrawable;
    }

    public void setButtonBackgroundDrawable(Drawable buttonBackgroundDrawable) {
        this.buttonBackgroundDrawable = buttonBackgroundDrawable;
    }

    public Drawable getVerifyButtonDrawable() {
        return verifyButtonDrawable;
    }

    public void setVerifyButtonDrawable(Drawable verifyButtonDrawable) {
        this.verifyButtonDrawable = verifyButtonDrawable;
    }

    public int getVerifyButtonWidthSize() {
        return verifyButtonWidthSize;
    }

    public int getVerifyButtonHeightSize() {
        return verifyButtonHeightSize;
    }

    public void setVerifyButtonWidthSize(int verifyButtonWidthSize) {
        this.verifyButtonWidthSize = verifyButtonWidthSize;
    }

    public void setVerifyButtonHeightSize(int verifyButtonHeightSize) {
        this.verifyButtonHeightSize = verifyButtonHeightSize;
    }

    public boolean isShowVerifyButton() {
        return showVerifyButton;
    }

    public void setShowVerifyButton(boolean showVerifyButton) {
        this.showVerifyButton = showVerifyButton;
    }

    public int getVerifyButtonPressesColor() {
        return verifyButtonPressesColor;
    }

    public void setVerifyButtonPressesColor(int verifyButtonPressesColor) {
        this.verifyButtonPressesColor = verifyButtonPressesColor;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/ui/enterpin/CustomizationOptionsBundle.kt
package co.tpcreative.supersafe.ui.enterpin
import android.graphics.drawable.Drawable

/**
 * The customization options for the buttons in [PinLockView]
 * passed to the [PinLockAdapter] to decorate the individual views
 *
 * Created by aritraroy on 01/06/16.
 */
class CustomizationOptionsBundle {
    private var textColor = 0
    private var textColorVerify = 0
    private var textSize = 0
    private var buttonSize = 0
    private var buttonBackgroundDrawable: Drawable? = null
    private var verifyButtonDrawable: Drawable? = null
    private var verifyButtonWidthSize = 0
    private var verifyButtonHeightSize = 0
    private var showVerifyButton = false
    private var verifyButtonPressesColor = 0
    fun getVerifyButtonNormalColor(): Int {
        return verifyButtonNormalColor
    }

    fun setVerifyButtonNormalColor(verifyButtonNormalColor: Int) {
        this.verifyButtonNormalColor = verifyButtonNormalColor
    }

    private var verifyButtonNormalColor = 0
    fun getTextColorVerify(): Int {
        return textColorVerify
    }

    fun setTextColorVerify(textColorVerify: Int) {
        this.textColorVerify = textColorVerify
    }

    fun getTextColor(): Int {
        return textColor
    }

    fun setTextColor(textColor: Int) {
        this.textColor = textColor
    }

    fun getTextSize(): Int {
        return textSize
    }

    fun setTextSize(textSize: Int) {
        this.textSize = textSize
    }

    fun getButtonSize(): Int {
        return buttonSize
    }

    fun setButtonSize(buttonSize: Int) {
        this.buttonSize = buttonSize
    }

    fun getButtonBackgroundDrawable(): Drawable? {
        return buttonBackgroundDrawable
    }

    fun setButtonBackgroundDrawable(buttonBackgroundDrawable: Drawable?) {
        this.buttonBackgroundDrawable = buttonBackgroundDrawable
    }

    fun getVerifyButtonDrawable(): Drawable? {
        return verifyButtonDrawable
    }

    fun setVerifyButtonDrawable(verifyButtonDrawable: Drawable?) {
        this.verifyButtonDrawable = verifyButtonDrawable
    }

    fun getVerifyButtonWidthSize(): Int {
        return verifyButtonWidthSize
    }

    fun getVerifyButtonHeightSize(): Int {
        return verifyButtonHeightSize
    }

    fun setVerifyButtonWidthSize(verifyButtonWidthSize: Int) {
        this.verifyButtonWidthSize = verifyButtonWidthSize
    }

    fun setVerifyButtonHeightSize(verifyButtonHeightSize: Int) {
        this.verifyButtonHeightSize = verifyButtonHeightSize
    }

    fun isShowVerifyButton(): Boolean {
        return showVerifyButton
    }

    fun setShowVerifyButton(showVerifyButton: Boolean) {
        this.showVerifyButton = showVerifyButton
    }

    fun getVerifyButtonPressesColor(): Int {
        return verifyButtonPressesColor
    }

    fun setVerifyButtonPressesColor(verifyButtonPressesColor: Int) {
        this.verifyButtonPressesColor = verifyButtonPressesColor
    }
}